annotationPipeline/src/main/java/org/cbioportal/annotation/AnnotationPipeline.java

```diff
@@ -114,20 +115,43 @@ public static void subMain(String[] args) throws NoSubcommandFoundException, Par
             subcommand = new AnnotateSubcommand(args);
         }
         if (subcommand instanceof AnnotateSubcommand) {
+            Instant start = Instant.now();
             annotate(subcommand, args);
+            System.out.println(" RUNTIME: " + Duration.between(start, Instant.now()).getSeconds() + " secs.");
         } else if (subcommand instanceof MergeSubcommand) {
             merge(subcommand);
+        } else if (subcommand instanceof VersionSubcommand) {
+            version(subcommand);
         }
     }
 
     public static void main(String[] args) {
-        Instant start = Instant.now();
         try {
             subMain(args);
         } catch (Exception e) {
             LOG.error(e.getMessage());
         }
-        System.out.println(" RUNTIME: " + Duration.between(start, Instant.now()).getSeconds() + " secs.");
+    }
+
+    private static String getAppVersion() {
+        java.io.InputStream is = AnnotationPipeline.class.getClassLoader().getResourceAsStream("maven.properties");
+        java.util.Properties p = new Properties();
+        try {
+            p.load(is);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        String appVersion = p.getProperty("app.version");
+
+        return appVersion;
+    }
```

Comment:
wasn't able to get this working with standard spring decorators, because it's all using static methods etc